### PR TITLE
Tq42 1267

### PR DESCRIPTION
**Pull request by lusnig**
Add block code

### CODE REVIEW

**Review by K1mpossible — DISMISSED** [review]
I did not review every hyperparameter for accuracy but the code blocks look generally correct. I see QuEnc already has API and CLI code samples in the docs so that is taken care of.

**Comment by lusnig** [comment]
This PR contain:

1. TQ42-575 add deep links
2. TQ42-1267 add block code

File: docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md [hunk 1]
```diff
@@ -42,3 +42,50 @@ The number of parents in a iteration
...
+ExperimentRun.create(
+    client=client, 
+    algorithm=AlgorithmProto.CVA_OPT, 
+    experiment_id=experiment_id,
+    compute=HardwareProto.SMALL, 
+    parameters={'parameters': cva_params, 'inputs': {} }
+)
+```
+
+```python
```
Comment (sbtq) [inline docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md:89]: why is there an empty python code block here?

File: docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md [hunk 2]
```diff
@@ -42,3 +42,50 @@ The number of parents in a iteration
 
 The number of offsprings created in a iteration
 
+## Example
```
Comment (sbtq) [inline docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md:45]: this is exactly the notebook code from cvaopt, right?
Could we maybe instead of having this code in here just reference the notebook? So we only have to adapt one place? Or was this a specific requirement from @K1mpossible ?

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md [hunk 3]
```diff
@@ -51,6 +101,49 @@ The Long Short-Term Memory (LSTM) network is a type of recurrent neural network
...
+    OptimProto, 
+    LossFuncProto, 
+    DatasetStorageInfoProto, 
+    TSLSTMTrainMetadataProto, 
+    TSLSTMTrainParametersProto,
+    MLTrainInputsProto
+)
+from google.protobuf.json_format import MessageToDict
+
+client = TQ42Client()
```
Comment (sbtq) [inline docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:120]: same as above

File: docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md [hunk 4]
```diff
@@ -42,3 +42,50 @@ The number of parents in a iteration
...
+```python
+from tq42.client import TQ42Client
+from tq42.experiment_run import ExperimentRun
+from tq42.algorithm import AlgorithmProto
+from tq42.compute import HardwareProto
+
+client = TQ42Client()
```
Comment (sbtq) [inline docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md:52]: same as mentioned in the other places

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md [hunk 5]
```diff
@@ -89,6 +182,69 @@ The Hybrid Quantum Multilayer Perceptron (HQMLP) represents an innovative leap f
...
+    DiffMethodProto,
+    QubitTypeProto,
+    DatasetStorageInfoProto, 
+    MLTrainInputsProto,
+    TSHQMLPTrainMetadataProto, 
+    TSHQMLPTrainParametersProto
+)
+from google.protobuf.json_format import MessageToDict
+
+client = TQ42Client()
```
Comment (sbtq) [inline docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:207]: same as above

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md [hunk 6]
```diff
@@ -89,6 +182,69 @@ The Hybrid Quantum Multilayer Perceptron (HQMLP) represents an innovative leap f
...
+    dropout_p=0.5,
+    bn=False,
+    num_epochs=20,
+    batch_size=32,
+    learning_rate=0.001,
+    loss_func=LossFuncProto.MAE,
+    optim=OptimProto.ADAM,
+    time_column="your_time_column",
+    target_column="your_target_column"
+
```
Comment (sbtq) [inline docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:234]: why these empty lines? Can we please properly format.

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md [hunk 7]
```diff
@@ -115,6 +271,51 @@ The Hybrid Quantum Long Short-Term Memory (HQLSTM) is an advanced version of the
...
+    OptimProto, 
+    LossFuncProto, 
+    DatasetStorageInfoProto, 
+    TSHQLSTMTrainMetadataProto, 
+    TSHQLSTMTrainParametersProto,
+    MLTrainInputsProto
+)
+from google.protobuf.json_format import MessageToDict
+
+client = TQ42Client()
```
Comment (sbtq) [inline docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:290]: same as above

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md [hunk 8]
```diff
@@ -115,6 +271,51 @@ The Hybrid Quantum Long Short-Term Memory (HQLSTM) is an advanced version of the
...
+client.login()
+
+params = MessageToDict(TSHQLSTMTrainMetadataProto(
+    parameters=TSHQLSTMTrainParametersProto(
+    input_width=20,
```
Comment (sbtq) [inline docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:295]: not properly indented

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md [hunk 9]
```diff
@@ -115,6 +271,51 @@ The Hybrid Quantum Long Short-Term Memory (HQLSTM) is an advanced version of the
...
+    num_epochs=20,
+    batch_size=32,
+    learning_rate=0.001,
+    depth=2,
+    n_qlayers=2,
+    num_qubits=2,
+    optim=OptimProto.ADAM,
+    loss_func=LossFuncProto.MAE
+    ),
+    inputs=MLTrainInputsProto(data=DatasetStorageInfoProto(storage_id="your_storage_id"))),
```
Comment (sbtq) [inline docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:308]: empty line? Little bit too much copy pasting?

File: docs/source/TetraOpt_Algorithm/Parameters_and_Hyperparameters.md [hunk 10]
```diff
@@ -19,4 +19,43 @@ Every problem will have a problem dimensionality determined by the domain of the
...
+            'iteration_number': 4, 
+            'maximal_rank': 4, 
+            'points_number': 1, 
+            'quantization': False,
+            'tolerance':0.0010000000474974513, 
+            'lower_limits': [0,0,0,0,0,0,0,0,0,0], 
+            'upper_limits': [9,9,9,9,9,9,9,9,9,9], 
+            'grid': [10,10,10,10,10,10,10,10,10,10],
+            'objective_function':'http://34.32.169.11:8000/test_func_eval/Ackley/'
+             #optional arguments below
```
Comment (sbtq) [inline docs/source/TetraOpt_Algorithm/Parameters_and_Hyperparameters.md:44]: that looks like really weird formatting

File: docs/source/TetraOpt_Algorithm/Parameters_and_Hyperparameters.md [hunk 11]
```diff
@@ -19,4 +19,43 @@ Every problem will have a problem dimensionality determined by the domain of the
...
+from tq42.client import TQ42Client
+from tq42.experiment_run import ExperimentRun
+from tq42.compute import HardwareProto
+from tq42.algorithm import AlgorithmProto
 
+client = TQ42Client()
+client.login()
+
+parameters = {
+        "parameters": {
```
Comment (sbtq) [inline docs/source/TetraOpt_Algorithm/Parameters_and_Hyperparameters.md:33]: why do use a dict here instead of a message like you were doing in all other examples?

File: docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md (hunk 1 above)
Comment (lusnig) [reply docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md:89]: removed it

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md [hunk 12]
```diff
@@ -28,6 +28,56 @@ The Multilayer Perceptron (MLP) is a staple in the realm of artificial neural ne
...
+    LossFuncProto, 
+    DatasetStorageInfoProto, 
+    MLTrainInputsProto,
+    TSMLPTrainMetadataProto, 
+    TSMLPTrainParametersProto,
+    ActFuncProto
+)
+from google.protobuf.json_format import MessageToDict
+
+client = TQ42Client()
```
[the remark this answers is not in the document]
Comment (lusnig) [reply docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:48]: done

File: docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md (hunk 4 above)
Comment (lusnig) [reply docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md:52]: done

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md [hunk 13]
```diff
@@ -51,6 +101,49 @@ The Long Short-Term Memory (LSTM) network is a type of recurrent neural network
...
+    label_width=1,
+    hidden_size=5,
+    dropout_coef=0.4,
+    num_epochs=20,
+    batch_size=32,
+    learning_rate=0.001,
+    optim=OptimProto.ADAM,
+    loss_func=LossFuncProto.MAE
+    ),
+    inputs=MLTrainInputsProto(data=DatasetStorageInfoProto(storage_id="your_storage_id"))),
```
[the remark this answers is not in the document]
Comment (lusnig) [reply docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:135]: removed

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md (hunk 5 above)
Comment (lusnig) [reply docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:207]: done

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md (hunk 9 above)
Comment (lusnig) [reply docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:308]: done

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md (hunk 7 above)
Comment (lusnig) [reply docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:290]: done

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md (hunk 3 above)
Comment (lusnig) [reply docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:120]: done

File: docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md (hunk 8 above)
Comment (lusnig) [reply docs/source/TQNet_Machine_Learning/Pre-Defined_Model_Architectures.md:295]: done

File: docs/source/TetraOpt_Algorithm/Parameters_and_Hyperparameters.md (hunk 10 above)
Comment (lusnig) [reply docs/source/TetraOpt_Algorithm/Parameters_and_Hyperparameters.md:44]: done

File: docs/source/TetraOpt_Algorithm/Parameters_and_Hyperparameters.md (hunk 11 above)
Comment (lusnig) [reply docs/source/TetraOpt_Algorithm/Parameters_and_Hyperparameters.md]: done

**Comment by lusnig** [comment]
Use a Message instead a python dict also in CVAOpt_Algorithm/Parameters_and_Hyperparameters.md

**Review by K1mpossible — APPROVED** [review]
I think these look correct.

**Review by K1mpossible — APPROVED** [review]

**Review by K1mpossible — APPROVED** [review]

**Review by sbtq — DISMISSED** [review]

**Review by sbtq — APPROVED** [review]
lgtm
